optaplanner-operator/optaplanner-operator-integration-test-parent/optaplanner-operator-integration-test/src/test/java/org/opaplanner/operator/integration/OptaPlannerSolverReconcilerIT.java

```diff
@@ -0,0 +1,140 @@
+package org.opaplanner.operator.integration;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.optaplanner.operator.impl.solver.model.ConfigMapDependentResource;
+import org.optaplanner.operator.impl.solver.model.OptaPlannerSolver;
+import org.optaplanner.operator.impl.solver.model.OptaPlannerSolverStatus;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+
+import io.fabric8.kubernetes.api.model.Condition;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.GenericKubernetesResource;
+import io.fabric8.kubernetes.api.model.Namespace;
+import io.fabric8.kubernetes.api.model.NamespaceBuilder;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+
+public class OptaPlannerSolverReconcilerIT {
+
+    public static final String ARTEMIS_BROKER_YAML = "artemis-broker.yaml";
+    public static final String SCHOOL_TIMETABLING_SOLVER_YML = "school-timetabling-solver.yml";
+    private final static KubernetesClient kubernetesClient = new DefaultKubernetesClient();
+    private static String solverName;
+    private static Namespace testNamespace;
+
+    @AfterAll
+    public static void tearDown() {
+        kubernetesClient.close();
+    }
+
+    @AfterEach
+    public void remove() {
+        kubernetesClient.namespaces().delete(testNamespace);
+    }
+
+    @BeforeEach
+    public void createNamespace() {
+        solverName = "school-timetabling-" + RandomStringUtils.randomNumeric(4);
+        testNamespace = kubernetesClient.namespaces()
+                .create(new NamespaceBuilder().withNewMetadata().withName(solverName).endMetadata().build());
+    }
+
+    @Test
+    public void optaPlannerSolverReconcilerIT() {
```

Comment:
We agreed to start by mimicking what the unit test does, which is now done.

Let's continue from this point further to implement the real integration test scenarios:
- make sure the deployment based on the solver image can start
- send some messages to the broker and check responses
- make sure the operator scales the deployment as expected according to the load we generate in the test

For these scenarios, I will drop some extra comments.